service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotificationHandler.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.events.SubmittedCaseEvent;
+import uk.gov.hmcts.reform.fpl.service.HmctsCourtLookUpService;
+import uk.gov.hmcts.reform.fpl.service.LocalAuthorityService;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class NotificationHandler {
+
+    private final HmctsCourtLookUpService hmctsCourtLookUpService;
+    private final LocalAuthorityService localAuthorityService;
+    private final NotificationClient notificationClient;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private String timeFramePresent;
```

Comment:
could this exist in the method? theres no public use of it among the class

Reply:
its a little neater up there but its used a few lines below its assignment

Reply:
Please remove field since it is used only by one method.

Presence of the field makes bean stateful and decreases cohesion.